### PR TITLE
Add puts_verbose

### LINKED ISSUE #24: Proposal: puts_verbose
It would be nice if we could standardize on a way to make the buildpacks log more information for debugging.

Relevant issues in the node buildpack:
https://github.com/heroku/heroku-buildpack-nodejs/issues/190
https://github.com/heroku/heroku-buildpack-nodejs/pull/398

`puts_verbose` would only log its output if an environment variable that we standardize on is set. Suggestion: `$BUILDPACK_VERBOSE`.

And I could also use this same variable to deliberately run `yarn` or `npm` in verbose mode.

Also allows support to have a standard response when builds are failing: run `heroku config:set BUILDPACK_VERBOSE=true` and rebuild.

### PR DESCRIPTION
**Pull request by jmorrell**
Standardize on a way to make the buildpacks log more information for debugging.

`puts_verbose` will only log its output if $BUILDPACK_VERBOSE is set. The idea is that we can use the same environment variable to opt-in to the verbose mode on any other commands that we run. This gives users one command to get more detailed logs, and makes it easier for support to direct users to turn on verbose mode.

```
heroku config:set BUILDPACK_VERBOSE=true
```

Fixes #24

### CODE REVIEW

**Comment by jmorrell** [comment]
Also, when it comes to bash

fa5

so thorough code review requested :)

File: puts_verbose [hunk 1]
```diff
@@ -0,0 +1 @@
+foo
```
Comment (kennethreitz) [inline puts_verbose:1]: is this file intended to be checked in?

**Review by kennethreitz — CHANGES_REQUESTED** [review]

File: stdlib.sh [hunk 2]
```diff
@@ -2,6 +2,7 @@
 # ---------------
 
 export BUILDPACK_LOG_FILE=${BUILDPACK_LOG_FILE:-/dev/null}
+export BUILDPACK_VERBOSE=${BUILDPACK_VERBOSE:-}
```
Comment (kennethreitz) [inline stdlib.sh:5]: I don't think this line is necessary.

**Review by kennethreitz — CHANGES_REQUESTED** [review]

**Comment by kennethreitz** [comment]
Will cleanup manually.

**Comment by kennethreitz** [comment]
We might want to add a mechanism for loading BUILDPACK_VERBOSE from the user env easily too.

**Comment by kennethreitz** [comment]
I'll open an issue for that.

**Comment by kennethreitz** [comment]
@jmorrell I also added `is_verbose`, for buildpack functionality.

File: puts_verbose (hunk 1 above)
Comment (jmorrell) [reply puts_verbose:1]: tenor
